test/e2e/karmadactl_test.go

```diff
@@ -1154,6 +1154,124 @@ var _ = ginkgo.Describe("Karmadactl options testing", func() {
 	})
 })
 
+var _ = framework.SerialDescribe("Karmadactl taint testing", func() {
+	ginkgo.Context("Test karmadactl taint command with different effects", func() {
+		var clusterName string
+		var taintKey = "environment"
+		var taintValue = "production"
+
+		ginkgo.BeforeEach(func() {
+			clusterName = framework.ClusterNames()[0]
+		})
+
+		// testTaintEffect is a reusable function for testing taints on a given effect.
+		testTaintEffect := func(effect, overwriteEffect corev1.TaintEffect) {
+			currentEffect := effect
+			taint := fmt.Sprintf("%s=%s:%s", taintKey, taintValue, currentEffect)
+			taintRemove := fmt.Sprintf("%s-", taint)
```

Comment:
```suggestion
			productionTaint := fmt.Sprintf("%s=%s:%s", taintKey, production, effect)
			testTaint := fmt.Sprintf("%s=%s:%s", taintKey, test, effect)
			taintRemove := fmt.Sprintf("%s-", testTaint)
```
The e2e failed because of the previous incorrect use of `--overwrite`, which was actually two taint, but ended up deleting only one of them, causing `membe1` to be unavailable in subsequent e2e tests because of the taint